web/src/date-utils.ts

```diff
@@ -54,31 +33,46 @@ export const dateToFormat = (date: Date | number, format: DateFormat): string =>
   if (isDateValid(date)) {
     switch (format) {
       case DateFormat.TimeShort:
-        return timeShortFormatter.format(date);
+        {
+          const dateObject = new Date(date);
+          const hours = padLeadingZero(dateObject.getHours());
+          return `${hours}:${dateObject.getMinutes()}`;
+        }
         break;
       case DateFormat.TimeMed:
-        return timeMedFormatter.format(date);
+        {
+          const dateObject = new Date(date);
+          const hours = padLeadingZero(dateObject.getHours());
+          return `${hours}:${dateObject.getMinutes()}:${dateObject.getSeconds()}`;
+        }
         break;
       case DateFormat.DateMed: {
-        const parts = dateMedFormatter.formatToParts(date);
-        return `${getPart(parts, 'year')}-${getPart(parts, 'month')}-${getPart(parts, 'day')}`;
+        const dateObject = new Date(date);
+        const month = padLeadingZero(dateObject.getMonth() + 1);
+
+        return `${dateObject.getFullYear()}-${month}-${dateObject.getDate()}`;
       }
       case DateFormat.TimeFull: {
-        const fractionalSeconds =
-          typeof date === 'number' ? Math.floor(date % 1000) : date.getMilliseconds();
-        return `${timeMedFormatter.format(date)}.${fractionalSeconds}`;
+        const dateObject = new Date(date);
+        const fractionalSeconds = padLeadingZero(dateObject.getMilliseconds(), 3);
+        const hours = padLeadingZero(dateObject.getHours());
+
+        return (
+          `${hours}:${dateObject.getMinutes()}` + `:${dateObject.getSeconds()}.${fractionalSeconds}`
```

Comment:
We should zero pad the minutes and seconds too. (Also in some of the other `case` branches.